Add played game badge

dop-amin opened the pull request:
- Add Night owl badge (🦉): Played a game between 8pm and 6am
- Add Early bird badge (🐦): Played a game between 6am and 9am
- Add Cat badge (🐱): Lost against majority of unique opponents
- Add Addict badge (💉): Played 21+ games in any 7-day period (3+ per day average)
- Fix tooltip bug where badge label was duplicated (e.g., "On fire: On fire")

All new badges are computed efficiently in bulk using the precompute_badge_data function to minimize database queries.

claude committed:
Add new achievement badges and fix tooltip redundancy

- Add Night owl badge (🦉): Played a game between 8pm and 6am
- Add Early bird badge (🐦): Played a game between 6am and 9am
- Add Cat badge (🐱): Lost against majority of unique opponents
- Add Addict badge (💉): Played 21+ games in any 7-day period (3+ per day average)
- Fix tooltip bug where badge label was duplicated (e.g., "On fire: On fire")

All new badges are computed efficiently in bulk using the precompute_badge_data function to minimize database queries.

Rixxc commented on templates/partials/leaderboard.html:
```diff
@@ -35,7 +35,7 @@
                               data-bs-toggle="tooltip"
                               data-bs-placement="top"
-                              title="{{ badge.label }}: {{ badge.tooltip }}">{{ badge.emoji }}</span>
+                              title="{{ badge.tooltip }}">{{ badge.emoji }}</span>
```
Now, the tooltip for doesn't follow the structure 'label: tooltip' anymore. There are badges where the label is not included in the tooltip. Instead, the redundant label information should be removed from the 'on fire' tooltip.

claude committed:
Fix tooltip structure to maintain consistent 'label: tooltip' format

Address PR review feedback: Instead of removing the label prefix from the
template globally, remove it only from the specific badge tooltips that had
redundant labels (Unstoppable, On Fire, Hot Streak).

This maintains the consistent "Label: tooltip" structure across all badges
while fixing the duplication issue (e.g., "On Fire: On Fire: ...").

- Revert template to show "{{ badge.label }}: {{ badge.tooltip }}"
- Remove redundant label prefixes from streak badge tooltips
- All badges now consistently display as "Label: Description"

Rixxc reviewed (approved):
looks good
